variables.tf

```diff
@@ -904,3 +904,16 @@ variable "custom_container_apps" {
   }))
   default = {}
 }
+
+variable "container_apps_infra_subnet_service_endpoints" {
+  description = "Endpoints to assign to infra subnet"
+  type        = list(string)
+  default     = []
+
+  validation {
+    condition = alltrue([
+      for endpoint in var.container_apps_infra_subnet_service_endpoints : contains(["Microsoft.Storage", "Microsoft.Sql", "Microsoft.AzureActiveDirectory", "Microsoft.AzureCosmosDB", "Microsoft.Web", "Microsoft.KeyVault", "Microsoft.EventHub", "Microsoft.ServiceBus", "Microsoft.ContainerRegistry", "Microsoft.CognitiveServices", "Microsoft.Storage.Global"], endpoint)
+    ])
+    error_message = "Service endpoints must match one of following: Microsoft.Storage Microsoft.Sql Microsoft.AzureActiveDirectory Microsoft.AzureCosmosDB Microsoft.Web Microsoft.KeyVault Microsoft.EventHub Microsoft.ServiceBus Microsoft.ContainerRegistry Microsoft.CognitiveServices Microsoft.Storage.Global"
```

Comment:
If we didn't include the `validation` block here, would the Provider respond with the same error anyway?

Reply:
I never even thought of even checking that, but surprisingly no the Provider doesn't seem to cover it

![allowed serviceendpoint](https://github.com/DFE-Digital/terraform-azurerm-container-apps-hosting/assets/131891854/bee6a65c-2f5b-48f6-864b-d401e5360e40)

Reply:
Ah. It won't show on `plan` but it will on `apply` because the underlying Azure API will reject the changes 🙃 Typical Azure 🙃